cryptolib/cryptolib.go

```diff
@@ -41,20 +61,67 @@ func HandleCryptoTicker(ctx context.Context, f *finnhub.DefaultApiService, s *di
 }
 
 // GetQuoteForCryptoAsset returns the TickerValue for Crypto Ticker.
-func GetQuoteForCryptoAsset(ctx context.Context, f *finnhub.DefaultApiService, asset string) (*messagelib.TickerValue, error) {
-	// Finnhub takes symbols in the format "GEMINI:btcusd"
-	formattedAsset := *cryptoExchange + ":" + strings.ToLower(asset) + "usd"
-	quote, _, err := f.CryptoCandles(ctx,
-		/* symbol= */ formattedAsset,
-		/* resolution= */ "1", // 1 = 1 hour
-		/* from= */ time.Now().Add(-1*time.Minute).Unix(),
-		/* to= */ time.Now().Unix())
-	if err != nil {
-		return nil, err
+func GetQuoteForCryptoAsset(ctx context.Context, asset string) (*messagelib.TickerValue, error) {
+	formattedAsset := strings.ToUpper(asset) + "USD"
+	priceFeeds, ok := ctx.Value(PriceFeeds).([]PriceFeed)
+	if !ok {
+		return nil, errors.New("couldn't unmarshal priceFeeds from context")
 	}
-	if len(quote.C) == 0 {
-		// A value of 0.0 means that the Ticker is Undefined.
+	priceFeed, ok := getPriceFeed(priceFeeds, formattedAsset)
+	if !ok {
 		return &messagelib.TickerValue{Ticker: asset, Value: 0.0, Change: 0.0}, nil
 	}
-	return &messagelib.TickerValue{Ticker: asset, Value: quote.C[0], Change: 0.0}, nil
+
+	price, err := strconv.ParseFloat(priceFeed.Price, 32)
+	if err != nil {
+		return &messagelib.TickerValue{Ticker: asset, Value: 0.0, Change: 0.0}, nil
+	}
+	change, err := strconv.ParseFloat(priceFeed.Change, 32)
+	if err != nil {
+		return &messagelib.TickerValue{Ticker: asset, Value: float32(price), Change: 0.0}, nil
+	}
+	return &messagelib.TickerValue{Ticker: asset, Value: float32(price), Change: float32(change)*100.0}, nil
+}
+
+func getPriceFeed(priceFeeds []PriceFeed, asset string) (*PriceFeed, bool) {
+	for i := range priceFeeds {
+		if priceFeeds[i].Pair == asset {
+			return &priceFeeds[i], true
+		}
+	}
+	return nil, false
+}
+
+// GetPriceFeeds returns the current Price points of cryptos traded on Gemini.
+func GetPriceFeeds() ([]PriceFeed, error) {
+	var priceFeeds []PriceFeed
+	url := geminiBaseURL + geminiPriceFeedURI
+	geminiClient := http.Client{
+		Timeout: time.Second * 2, // Timeout after 2 seconds
+	}
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return priceFeeds, err
+	}
+
+	req.Header.Set("User-Agent", "brokerbot")
+
+	res, getErr := geminiClient.Do(req)
+	if getErr != nil {
+		return priceFeeds, getErr
```

Comment:
Convention is to return nil, err upon error. Passing the empty priceFeeds var is uneccesary.